rfc/0077-zone-aware-replica-reads.md

```diff
@@ -0,0 +1,183 @@
+# Meta
+
+| Field          | Value                    |
+|----------------|--------------------------|
+| RFC Name       | Zone-Aware Replica Reads |
+| RFC ID         | 77                       |
+| Start Date     | 2024-04-25               |
+| Owner          | Sergey Avseyev           |
+| Current Status | DRAFT                    |
+| Revision       | #1                       |
+
+# Summary
+
+This document describes changes that SDKs must implement to optimize costs of
+network traffic when the server is being deployed across multiple availability
+zones.
+
+# Motivation
+
+Existing SDKs do not take into account node availability groups of the nodes in
+the cluster. In particular, when performing `getAllReplicas` operations, the SDK
+would send `GET (0x00)` request to active vBucket and `GET_REPLICA (0x83)` to
+each of the replicas. This approach will make reading from replicas expensive
+operations if nodes of the cluster deployed in different availability zones
+(AZ).
+
+# Use Cases
+
+## No Preference
+
+This is current behaviour, where the SDK does not take into account server
+groups.
+
+![figure 1: No Prefrence](figures/0077-case-1-no-preference.svg)
+
+## Local Only
+
+In this case, all reads will be done only from the local availability zone. It
+is the cheapest solution, although it reduces chances of getting data from
```

Comment:
increases chances surely?

Reply:
maybe I incorrectly stated it, but it is related to one of the questions later. Where by selecting string Local-Only strategy we might be in the situation, where selected group does not have any replica, and maybe not even master copy of the vbucket.

Reply:
Oh I see.  So it reduces chances of getting data, full stop.  I took it as meaning a replica instead of the active.